pycaret/tests/test_time_series.py

```diff
@@ -59,25 +79,31 @@ def load_ts_models(load_setup):
     return ts_estimators
 
 
-@pytest.mark.parametrize("fh, model", _parametrize_create_model)
-def test_create_model(fh, model, load_data):
+###########################
+#### Fixtures End Here ####
+###########################
 
-    from pycaret.internal.pycaret_experiment import TimeSeriesExperiment
+##########################
+#### Tests Start Here ####
+##########################
 
+def test_create_model(load_setup, model_parameters, load_data):
+    """test create_model functionality
+    """
     exp = TimeSeriesExperiment()
+    for name, fh in model_parameters:
```

Comment:
I think that here when pytest collects the test it would not report the tests as separate

```
test_time_series.py::test_create_model[8-ARIMA]                   
test_time_series.py::test_create_model[6-Polynomial]
test_time_series.py::test_create_model[7-ETS]
```
maybe there's a way to `yield` the results from fixtures or passing the results from the fixture throught the `argvalues` argument of `pytest.mark.parametrize`. 

Some references: [here](https://docs.pytest.org/en/stable/reference.html?highlight=pytest%20mark%20parametrize#pytest.python.Metafunc.parametrize), [here](https://stackoverflow.com/questions/52764279/pytest-how-to-parametrize-a-test-with-a-list-that-is-returned-from-a-fixture) and [here](https://stackoverflow.com/questions/50231627/python-pytest-unpack-fixture/56865893)

Reply:
Thanks @TremaMiguel , Yes, it would be nice to convert it to a parameterized test. I will experiment more with it before the next PR. Thanks for the references.

Reply:
Looks like one can pass a function to `pytest.mark.parametrize()` , for example,

```python
def model_parameters():

    globals_dict = {"seed": 0, "n_jobs_param": -1}
    model_containers = get_all_model_containers(globals_dict)

    model_names_ = []
    for model_name in model_containers.keys():
        if not model_name.startswith(("ensemble")):
            model_names_.append(model_name)

    parameters = [
        (name, np.arange(1, randint(6, 24)) if i%2==0  else randint(6, 24))
        for i, name in enumerate(model_names_)
    ]

    return parameters
```

this function can be called inside `test_create_model` 

```python
@pytest.mark.parametrize("model_parameter", model_parameters())
def test_create_model(load_setup, model_parameter, load_data):
    """test create_model functionality
    """
    exp = TimeSeriesExperiment()
    name, fh = model_parameter
    # Need to create individual setup for each model since the `fh` will be different for all models
    exp.setup(
        data=load_data, fold=3, fh=fh, fold_strategy="expandingwindow", verbose=False
    )
    model_obj = exp.create_model(name)

    y_pred = model_obj.predict()
    assert isinstance(y_pred, pd.Series)

    fh_index = fh if isinstance(fh, int) else fh[-1]
    expected_period_index = load_data.iloc[-fh_index:].index
    assert np.all(y_pred.index == expected_period_index)
```

will be adding this in my next PR.